packages/docusaurus-theme-classic/src/theme/SearchModal/index.tsx

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState} from 'react';
+import styles from './style.module.css';
+import {useBaseUrlUtils} from '@docusaurus/useBaseUrl';
+import Link from '@docusaurus/Link';
+import {useSearch} from '@theme/useSearch';
+import Fuse from 'fuse.js';
+import {TrySearchIcon, NoResultIcon} from './icon';
+
+const ALLOWED = 30;
+const formatString = (match: Fuse.FuseResultMatch, str: string) => {
+  const [start, end] = match.indices[0];
+  const length = end - start;
+  const strChunks = [
+    str.substr(0, start),
+    str.substr(start, length + 1),
+    str.substr(end + 1),
+  ];
+  if (str.length < ALLOWED) {
+    return (
+      <span>
+        {strChunks[0]}
+        <span className={styles.highlight}>{strChunks[1]}</span>
+        {strChunks[2]}
```

Comment:
try to make the code readable, assign a proper variable name to strChunks[2] otherwise I don't know what it contains :D